src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -338,9 +338,9 @@ class PrisonerIepLevelReviewService(
     }
   }
 
-  private suspend fun updateIepLevelWithCurrentFlagToFalse(bookingId: Long) {
-    prisonerIepLevelRepository.findFirstByBookingIdAndCurrentIsTrueOrderByReviewTimeDesc(bookingId)
-      ?.let {
+  private suspend fun updateIepLevelsWithCurrentFlagToFalse(bookingId: Long) {
+    prisonerIepLevelRepository.findAllByBookingIdInAndCurrentIsTrueOrderByReviewTimeDesc(listOf(bookingId))
+      .collect {
         prisonerIepLevelRepository.save(it.copy(current = false))
```

Comment:
Just out of curiosity, there is no update method? Is that why we use `copy()`? (When I tried to look up the tree I couldn't find one)

Reply:
Indeed - see `CoroutineCrudRepository`